kiwi_lint/generic_foreign_key.py

```diff
@@ -0,0 +1,22 @@
+from pylint.checkers import utils, BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+
+class GenericForeignKeyChecker(BaseChecker):
+    __implements__ = (IAstroidChecker, )
+
+    name = 'generic-foreign-key-checker'
+
+    msgs = {
+        'E4851': (
+            'Avoid using GenericForeignKey fields',
+            'avoid-generic-foreign-key',
+            'Avoid defining GenericForeignKey fields in models,'
+            'See: https://github.com/kiwitcms/Kiwi/issues/1303',
+        )
+    }
+
+    @utils.check_messages('avoid-generic-foreign-key')
+    def visit_call(self, node):
+        if node.func.as_string() == 'GenericForeignKey':
```

Comment:
.as_string() returns `GenericForeignKey(fk_field='object_pk')"` so your comparison will not work as-is.

Reply:
when I was trying this out and when I print it out, I get just `GenericForeignKey`

Reply:
Not sure why the difference but better use .find() to locate this string as part of a function call